#! /bin/sh /usr/share/dpatch/dpatch-run
## 23exitcode_q_ls.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: Make the exitcode of '-q -ls' and friends ('-qr -ls' and even
## DP: '-qdr -ls') match the documentation.

@DPATCH@
diff -Naur screen-4.0.3.orig/screen.c screen-4.0.3/screen.c
--- screen-4.0.3.orig/screen.c	2008-06-01 23:41:11.000000000 +0200
+++ screen-4.0.3/screen.c	2008-06-01 23:44:59.000000000 +0200
@@ -1073,8 +1073,12 @@
       eff_uid = real_uid;
       eff_gid = real_gid;
       i = FindSocket((int *)NULL, &fo, &oth, SockMatch);
-      if (quietflag)
-        exit(8 + (fo ? ((oth || i) ? 2 : 1) : 0) + i);
+      if (quietflag) {
+        if (rflag)
+          exit(10 + i);
+        else
+          exit(9 + (fo || oth ? 1 : 0) + fo);
+      }
       if (fo == 0)
         Panic(0, "No Sockets found in %s.\n", SockPath);
       Panic(0, "%d Socket%s in %s.\n", fo, fo > 1 ? "s" : "", SockPath);
